### PR TITLE
[flutter_tools] pin transitive deps during --transitive-closure

### LINKED ISSUE #80909: Analyze is failing, Warning: transitive closure contained non-allowlisted packages: [frontend_server_client]
dart --enable-asserts ./dev/bots/analyze.dart
```
▌17:09:37▐ Package Allowlist...
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Warning: transitive closure contained non-allowlisted packages:
[frontend_server_client]
See dev/bots/allowlist.dart for instructions on how to update the package allowlist.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

I see this on 07fc1a6a48 but I don't think this is related to a framework commit...

### PR DESCRIPTION
**Pull request by jonahwilliams**
This fixes the existing analysis test. Fixes https://github.com/flutter/flutter/issues/80909

When printing the transitive closure, ensure that all dependencies - even transitive dependencies - are pinned in the synthetic pubspec. This prevents us from accidentally grabbing new versions of transitive deps without a roll into the flutter tool.

This was only a problem with --transitive-closure, for update-packages --force-upgrade it is expected that we leave transitive deps unpinned to reduce the constraints for a new version solve

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -1269,12 +1269,12 @@ File _pubspecFor(Directory directory) {
...
 /// Generates the source of a fake pubspec.yaml file given a list of
 /// dependencies.
-String _generateFakePubspec(Iterable<PubspecDependency> dependencies) {
+String _generateFakePubspec(Iterable<PubspecDependency> dependencies, bool transitiveClosure) {
   final StringBuffer result = StringBuffer();
   final StringBuffer overrides = StringBuffer();
   result.writeln('name: flutter_update_packages');
   result.writeln('environment:');
-  result.writeln("  sdk: '>=2.10.0 <3.0.0'");
+  result.writeln("  sdk: '>=2.12.0 <3.0.0'");
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/update_packages.dart:1277]: while I'm at it...

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by Hixie** [comment]
test-exempt: clearly this is already tested since the tests failed ;-)

**Comment by jonahwilliams** [comment]
Actually, I did this wrong - but the tests are still failing thankfully :D

**Comment by jonahwilliams** [comment]
Actually rewrote the logic a bit so for --transitive-closure we only look at the deps of  the current packages instead of the generated package.

**Review by gspencergoog — APPROVED** [review]
LGTM still, with the changes.

**Pull request merged**